lib/src/sql/array.rs

```diff
@@ -317,13 +318,18 @@ pub trait Uniq<T> {
 
 impl Uniq<Array> for Array {
 	fn uniq(mut self) -> Array {
-		for x in (0..self.len()).rev() {
-			for y in (x + 1..self.len()).rev() {
-				if self[x] == self[y] {
-					self.remove(y);
-				}
+		let mut set: HashSet<&Value> = HashSet::new();
+		let mut to_remove: Vec<usize> = Vec::new();
+		for i in 0..self.len() {
+			if set.contains(&self[i]) {
+				to_remove.push(i);
+			} else {
+				set.insert(&self[i]);
 			}
 		}
+		for i in to_remove.iter().rev() {
+			self.remove(*i);
```

Comment:
Can we get away with `swap_remove` here since we are iterating in reverse?

Reply:
I was opposed to using `swap_remove` because I wanted to preserve order. But if we think order shouldn't matter for this, then yeah this works.

Reply:
Fair enough, preserving order is a nice property :)

Reply:
It's just that, in the worst case (array of **n** equal items), the time complexity will be still be **O(n<sup>2</sup>)**

Using [Abolish](https://github.com/surrealdb/surrealdb/blob/d5a378b1beae5d0ecee01e74bd89e15921cc5cb6/lib/src/sql/array.rs#L204-L224) where the closure does a lookup in `to_remove` would probably preserve order and be faster in the worst case due to the inner workings of [`retain`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain).